src/OmniSharp.Abstractions/Models/v1/TypeLookup/DocumentationComment.cs

```diff
@@ -166,6 +166,16 @@ private static string GetCref(string cref)
             }
             return cref + " ";
         }
+
+        private static string RetainSingleLeadingSpace(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+                return string.Empty;
+            if (!Char.IsWhiteSpace(input[0]))
+                return input;
+            int offset = input.TakeWhile(c => char.IsWhiteSpace(c)).Count();
+            return " " + input.Substring(offset);
```

Comment:
The use of LINQ caught my eye here. It looks like this will be a pretty hot path so we might do a little perf tuning.

Would deleting the LINQ expression and doing something like this satisfy the goal?

```csharp
return $" {input.TrimStart()}";
```

Reply:
FWIW, we allocate one string fewer by not calling TrimStart.

Reply:
Oops, @jonsequitur is right--I don't know why I thought string.substring didn't allocate a string. I like his solution.